rest/server_context.go

```diff
@@ -2021,21 +2013,22 @@ func (sc *ServerContext) Database(ctx context.Context, name string) *db.Database
 func (sc *ServerContext) initializeCouchbaseServerConnections(ctx context.Context, failFast bool) error {
```

Comment:
It looks like initializeCouchbaseServerConnections does two things:
1. Set up gocb connections for GoCBAgent admin auth (GoCBAgent, NoX509HTTPClient)
2. Set up the bootstrap connection
Now that the second one of these isn't Couchbase Server-specific, I feel like it might be cleaner to split into two functions (InitializeAdminAuthConnections, InitializeBootstrap), so that we can keep the 'ServerIsWalrus' check up in SetupServerContext.  This also avoids the cognitive dissonance of calling 'initializeCouchbaseServerConnections' to set up a rosmar bootstrap.